### PR TITLE
Layout Builder Compatibility

### PR DESCRIPTION
**Pull request by Sardis93**
Check not only for Paragraph but Block during Virtual Y Login Rediret, ensuring compatibility with Layout Builder.

**Related Issue/Ticket:**

**PLEASE CHECK BASE BRANCH FOR YOUR PR**
**ONLY urgent and approved fixes for point release should go to master branch**

(Replace this line with a link to the related GitHub issue on [YCloudYUSA/yusaopeny_gated_content](https://github.com/YCloudYUSA/yusaopeny_gated_content/issues) or your own ticketing system)

## Steps to test:

- [ ] Enable Layout Builder module.
- [ ] Enable openy_gc_auth_example module.
- [ ] Create Content Type for Layout Builder.
- [ ] Create node and add **Virtual Y Content** Block via Layout Builder.
- [ ] Create another node with **Virtual Y Login** Block via Layout Builder. 
- [ ] Set up **Virtual Y Landing Page url** and **Virtual Y Login Landing Page url** fields on settings page: _admin/openy/virtual-ymca_
- [ ] Navigate to the page with **Virtual Y Content Login** block and hit **Enter Virtual Y** button.
- [ ] Observe redirect to the node with **Virtual Y Content** Block
 
## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [x] This PR  provides updates via `hook_update_N` or other means.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

File: config/install/block_content.type.gated_content_login.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+langcode: en
+status: true
+dependencies: {  }
+id: gated_content_login
+label: 'Virtual Y Content Login'
+revision: 0
+description: 'A block to display Virtual Y Content login paragraph.'
```
Comment (anpolimus) [inline config/install/block_content.type.gated_content_login.yml:7]:
```suggestion
description: 'A block to display Virtual Y Content login.'
```

**Review by anpolimus — CHANGES_REQUESTED** [review]
Thanks for your approach, but this requests needs to be changed.
We are trying to get rid of paragraphs in the future.
This means that having a paragraph inside the block is not the way how we want to render VY in the future.
We need a simple custom block that renders an app inside it's template.
It will make your check code a little bit complex but module will be less dependant on paragraphs.

**Review by anpolimus — CHANGES_REQUESTED** [review]
@Sardis93 in this project we try to create a new block for each widget since we want to have a base for future improvements and comfortable use on the LB custom block level.
Please, return VY block back.

**Review by anpolimus — APPROVED** [review]

File: src/Plugin/Block/VirtualYApp.php [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $block_config = $this->getConfiguration();
+
+    $app_config = $this->gatedContentConfig->getOriginal();
+    // Give ability for 3rd party modules to alter data for the js app.
+    $this->moduleHandler->alter('virtual_y_app_settings', $app_config);
```
Comment (podarok) [inline src/Plugin/Block/VirtualYApp.php:234]: Let's have this documented in modulename.api.php file, please

**Review by podarok — CHANGES_REQUESTED** [review]
Small thing to get merged

**Comment by Sardis93** [comment]
Hey, @podarok !

The alter hook you suggest to document is not a new hook i added, it's the one we used from before and it is already in the api.php file. Please, see the link:

https://github.com/YCloudYUSA/yusaopeny_gated_content/blob/master/openy_gated_content.api.php#L22

**Pull request merged**